service.rb

```diff
@@ -7,49 +7,42 @@ class Service
   def initialize(port)
     app = -> environment {
       request = Rack::Request.new(environment)
+      response = Rack::Response.new
       store = YAML::Store.new("daily_data.yml")
 
       if request.get? && request.path == "/show/data"
-        content_type = "text/html"
-        status = 200
-        response_message = "<ul>\n"
+        response.write("<ul>\n")
+        response.content_type = "text/html; charset=UTF-8"
 
         all_data = {}
         store.transaction do
           all_data = store[:all_data]
         end
 
         all_data.each do |daily_data|
-          response_message << "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
+          response.write "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
         end
 
-        response_message << "</ul>\n"
-        response_message << daily_data_form
+        response.write "</ul>\n"
+        response.write daily_data_form
 
       elsif request.post? && request.path == "/add/data"
-        content_type = "text/html"
-        status = 303
-        response_message = ""
+        response.write("")
+        response.content_type = "text/html; charset=UTF-8"
+        response.redirect('/show/data')
```

Comment:
Extra credit: this used to be a [`303 See Other`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303), but now it’s become a [`302 Found`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/302) (extra extra credit: what’s the difference?); can you make it a `303 See Other` again?

Reply:
`302 found` is used for URL redirection, rather than another HTTP request. In this case, I want the page to `GET /show/data` and not specifically to another URL, so I think `303 See Other` is a better status to return.

I think we can just set the response status as so `response.status = 303`

Reply:
OK, but what about `#redirect`?

Reply:
`#redirect('/show/data', status: 303)` 🎉